src/game/Spell.cpp

```diff
@@ -5082,9 +5082,8 @@ SpellCastResult Spell::CheckCast(bool strict)
                 Creature* pet = m_caster->GetPet();
                 if (!pet)
                 {
-                    Pet* dbPet = new Pet;
-                    SpellCastResult result = dbPet->TryLoadFromDB((Player*)m_caster, 0);
-                    delete dbPet;
+                    Pet dbPet;
+                    SpellCastResult result = dbPet.TryLoadFromDB((Player*)m_caster, 0);
                     if(result != SPELL_FAILED_TARGETS_DEAD)
```

Comment:
That is weird, it just did on my end, sorry.

`if (dbPet.TryLoadFromDB((Player*)m_caster, 0) != SPELL_FAILED_TARGETS_DEAD)`

would be what I would have suggested in that case instead (again, since you don't use result for anything outside the scope)